src/software/ai/navigator/path_planner/simulated_theta_star_test.cpp

```diff
@@ -0,0 +1,255 @@
+    // The two points are not in the same coordinated initially, however after
+    // initial_position is moved to a free point, it is in the same coordinate as
+    // destination. Values are chosen from a case which previously did not find a path.
+    Point destination      = Point(1, 0);
+    Point initial_position = Point(0.96905413818359376, -0.26277551269531252);
+    setBallState(BallState(Point(0, -0.6), Vector(0, 0)));
+    addFriendlyRobots(
+        TestUtil::createStationaryRobotStatesWithId({Point(-3, 0), initial_position}));
+    addEnemyRobots(TestUtil::createStationaryRobotStatesWithId({Point(1, 0)}));
+
+    auto tactic = std::make_shared<MoveTactic>(false);
+    tactic->updateControlParams(destination, Angle::zero(), 0);
+    setTactic(tactic);
+    setRobotId(1);
+
+    std::set<MotionConstraint> motion_constraints;
+    motion_constraints.insert(MotionConstraint::ENEMY_ROBOTS_COLLISION);
+    setMotionConstraints(motion_constraints);
+
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [destination, tactic](std::shared_ptr<World> world_ptr,
+                              ValidationCoroutine::push_type& yield) {
+            // Small rectangle around the destination point that the robot should be
+            // stationary within for 15 ticks
+            Rectangle expected_final_position(Point(0.985, -0.26), Point(1.15, -0.3));
+            robotStationaryInPolygon(1, expected_final_position, 15, world_ptr, yield);
+        }};
+
+    std::vector<ValidationFunction> non_terminating_validation_functions = {};
+
+    runTest(terminating_validation_functions, non_terminating_validation_functions,
+            Duration::fromSeconds(15));
+}
+
+TEST_F(SimulatedThetaStarTest, test_theta_star_dest_in_obstacle)
+{
+    // Destination is in obstacle, but initial point is open
+    Point destination      = Point(1, -0.1);
+    Point initial_position = Point(1, 2);
+    setBallState(BallState(Point(0, -0.6), Vector(0, 0)));
+    addFriendlyRobots(
+        TestUtil::createStationaryRobotStatesWithId({Point(-3, 0), initial_position}));
+    addEnemyRobots(TestUtil::createStationaryRobotStatesWithId({Point(1, 0)}));
+
+    auto tactic = std::make_shared<MoveTactic>(false);
+    tactic->updateControlParams(destination, Angle::zero(), 0);
+    setTactic(tactic);
+    setRobotId(1);
+
+    std::set<MotionConstraint> motion_constraints;
+    motion_constraints.insert(MotionConstraint::ENEMY_ROBOTS_COLLISION);
+    setMotionConstraints(motion_constraints);
+
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [destination, tactic](std::shared_ptr<World> world_ptr,
+                              ValidationCoroutine::push_type& yield) {
+            // Small rectangle around the destination point that the robot should be
+            // stationary within for 15 ticks
+            Rectangle expected_final_position(Point(0.985, -0.26), Point(1.15, -0.3));
```

Comment:
this comment isn't accurate since the rectangle doesn't contain the destination

Reply:
Good point, I have updated it.